File: Calculadora.java
import java.util.Scanner; // Importamos esta herramienta para poder leer lo que el usuario teclea

public class Calculadora {

    // 1. DELEGACIÓN DE TAREAS EN FUNCIONES INDEPENDIENTES (Aquí se llaman "Métodos")
    // Usamos 'static' para poder usarlas directamente en el flujo principal
    public static double sumar(double a, double b) {
        return a + b;
    }

    public static double restar(double a, double b) {
        return a - b;
    }

    public static double multiplicar(double a, double b) {
        return a * b;
    }

    // Usamos 'void' porque esta función imprime el resultado ella misma y no regresa un dato
    public static void dividir(double a, double b) {
        // Estructura condicional para prevenir la división por cero
        if (b == 0) {
            System.out.println("Error: No se puede dividir entre cero.");
        } else {
            System.out.println("Resultado de la división: " + (a / b));
        }
    }

    // 2. FUNCIÓN PRINCIPAL (El punto de partida obligatorio en todo programa Java)
    public static void main(String[] args) {
        // Creamos un objeto 'scanner' para recibir datos desde la consola
        Scanner scanner = new Scanner(System.in);
        int opcion;
        double num1, num2;

        // Bucle continuo para prevenir el reinicio abrupto
        while (true) {
            System.out.println("\n=== CALCULADORA MULTILINGÜE (Java) ===");
            System.out.println("1. Sumar");
            System.out.println("2. Restar");
            System.out.println("3. Multiplicar");
            System.out.println("4. Dividir");
            System.out.println("5. Salir del programa");
            System.out.print("Selecciona una opción (1-5): ");
            
            opcion = scanner.nextInt(); // Lee un número entero

            // Condicional para romper el bucle y salir de forma segura
            if (opcion == 5) {
                System.out.println("Saliendo de la aplicación. ¡Hasta luego!");
                break;
            }

            // Estructuras condicionales para discriminar la operación deseada
            if (opcion >= 1 && opcion <= 4) {
                System.out.print("Ingresa el primer número: ");
                num1 = scanner.nextDouble(); // Lee un número decimal
                System.out.print("Ingresa el segundo número: ");
                num2 = scanner.nextDouble();

                if (opcion == 1) {
                    System.out.println("Resultado de la suma: " + sumar(num1, num2));
                } else if (opcion == 2) {
                    System.out.println("Resultado de la resta: " + restar(num1, num2));
                } else if (opcion == 3) {
                    System.out.println("Resultado de la multiplicación: " + multiplicar(num1, num2));
                } else if (opcion == 4) {
                    dividir(num1, num2);
                }
            } else {
                System.out.println("Opción no válida. Por favor, selecciona un número del 1 al 5.");
            }
        }
        
        scanner.close(); // Cerramos el lector para liberar memoria
    }
}

File: calculadora.cpp
#include <iostream>

using namespace std;

// 1. DELEGACIÓN DE TAREAS EN FUNCIONES INDEPENDIENTES
double sumar(double a, double b) {
    return a + b;
}

double restar(double a, double b) {
    return a - b;
}

double multiplicar(double a, double b) {
    return a * b;
}

// Estructura condicional interna para prevenir la división por cero
void dividir(double a, double b) {
    if (b == 0) {
        cout << "Error: No se puede dividir entre cero." << endl;
    } else {
        cout << "Resultado de la division: " << (a / b) << endl;
    }
}

// Función principal donde corre el flujo del programa
int main() {
    int opcion;
    double num1, num2;

    // 2. BUCLE CONTINUO
    while (true) {
        cout << "\n=== CALCULADORA MULTILINGUE (C++) ===" << endl;
        cout << "1. Sumar" << endl;
        cout << "2. Restar" << endl;
        cout << "3. Multiplicar" << endl;
        cout << "4. Dividir" << endl;
        cout << "5. Salir del programa" << endl;
        cout << "Selecciona una opcion (1-5): ";
        cin >> opcion;

        // Condicional para salir del bucle continuo de forma segura
        if (opcion == 5) {
            cout << "Saliendo de la aplicacion. ¡Hasta luego!" << endl;
            break; 
        }

        // Estructuras condicionales para discriminar la operacion deseada
        if (opcion >= 1 && opcion <= 4) {
            cout << "Ingresa el primer numero: ";
            cin >> num1;
            cout << "Ingresa el segundo numero: ";
            cin >> num2;

            if (opcion == 1) {
                cout << "Resultado de la suma: " << sumar(num1, num2) << endl;
            } else if (opcion == 2) {
                cout << "Resultado de la resta: " << restar(num1, num2) << endl;
            } else if (opcion == 3) {
                cout << "Resultado de la multiplicacion: " << multiplicar(num1, num2) << endl;
            } else if (opcion == 4) {
                dividir(num1, num2);
            }
        } else {
            cout << "Opcion no valida. Por favor, selecciona un numero del 1 al 5." << endl;
        }
    }

    return 0;
}
